### PR TITLE
Integrate NLTK with Tika fix for TIKA-1876 contributed by manalishah

### PR DESCRIPTION
**Pull request by manalishah**
Hi all,

Apache Tika already performs Named Entity Recognition using Open NLP and Stanford Core NLP. Natural Language Toolkit is another open source python library and I believe it will be a great idea to have NLTK integrated along with Tika.
NLTK can extract NER as well as classify them. For this purpose I, along with Prof Chris Mattmann have published NLTKRest, a python pip/setuptools installable module that exposes NLTK as a REST service.
https://github.com/manalishah/NLTKRest.git

https://wiki.apache.org/tika/TikaAndNLTK

### CODE REVIEW

File: tika-parsers/src/main/java/org/apache/tika/parser/ner/nltk/NLTKNERecogniser.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+ */
+package org.apache.tika.parser.ner.nltk;
+
+import org.apache.tika.parser.ner.NERecogniser;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
```
Comment (chrismattmann) [inline tika-parsers/src/main/java/org/apache/tika/parser/ner/nltk/NLTKNERecogniser.java]: no star imports, please fix to specific imports

File: tika-parsers/src/main/java/org/apache/tika/parser/ner/nltk/NLTKNERecogniser.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import java.util.*;
```
Comment (chrismattmann) [inline tika-parsers/src/main/java/org/apache/tika/parser/ner/nltk/NLTKNERecogniser.java]: no star imports, please fix to specific imports

File: tika-parsers/src/main/java/org/apache/tika/parser/ner/nltk/NLTKNERecogniser.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+    /**
+     * recognises names of entities in the text
+     * @param text text which possibly contains names
+     * @return map of entity type -> set of names
+     */
+    public Map<String, Set<String>> recognise(String text) {
+        Map<String, Set<String>> entities = new HashMap<>();
+        try {
+            int port = 8881;
```
Comment (chrismattmann) [inline tika-parsers/src/main/java/org/apache/tika/parser/ner/nltk/NLTKNERecogniser.java]: unused variable, remove

File: tika-parsers/src/test/java/org/apache/tika/parser/ner/nltk/NLTKNERecogniserTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        Metadata md = new Metadata();
+        tika.parse(new ByteArrayInputStream(text.getBytes(StandardCharsets.UTF_8)), md);
+
+        Set<String> names = new HashSet<>(Arrays.asList(md.getValues("NER_NAMES")));
+        if(names.size() == 0) {
+            return;
+        }
+        else {
+            assertTrue(names.contains("America"));
+            assertTrue(names.size() == 1); //and nothing else
```
Comment (chrismattmann) [inline tika-parsers/src/test/java/org/apache/tika/parser/ner/nltk/NLTKNERecogniserTest.java]: unneeded comment, please remove.

**Comment by chrismattmann** [comment]
looks great @manalishah please review and update per my comments. Thanks!

**Comment by manalishah** [comment]
@chrismattmann thanks! I have made the updates as per your comments.

**Comment by chrismattmann** [comment]
fantastic job @manalishah . I will merge today!

**Pull request merged**